Content.Shared/_RMC14/Xenonids/Construction/SharedXenoConstructionSystem.cs

```diff
@@ -987,6 +1010,112 @@ private void OnDeleteXenoResinHit(Entity<DeleteXenoResinOnHitComponent> ent, ref
             QueueDel(args.Target);
     }
 
+    private void OnXenoDoorBeforeClose(Entity<ResinDoorComponent> resinDoor, ref BeforeDoorClosedEvent args)
+    {
+        if (args.Partial)
+        {
+            return;
+        }
+        if (!TryComp<DoorComponent>(resinDoor, out var doorComp))
+        {
+            return;
+        }
+
+        List<EntityUid> bodies = new();
```

Comment:
This should be a field that gets cleared before use, so it doesn't create a new list every time